core/integration/features/liquidity-provision/automatic-adjustment-to-order-offsets-when-ref-price-outside-bounds.feature

```diff
@@ -0,0 +1,157 @@
+    Given the parties deposit on asset's general account the following amount:
+      | party | asset | amount                 |
+      | aux   | USD   | 1000000000000          |
+      | aux2  | USD   | 1000000000000          |
+      | lp    | USD   | 1000000000000000000000 |
+
+    When the parties submit the following liquidity provision:
+      | id  | party | market id | commitment amount | fee | side | pegged reference | proportion | offset | lp type    |
+      | lp1 | lp    | ETH/DEC19 | 90000             | 0.1 | buy  | MID              | 50         | 20     | submission |
+      | lp1 | lp    | ETH/DEC19 | 90000             | 0.1 | sell | MID              | 50         | 16     | submission |
+      | lp1 | lp    | ETH/DEC19 | 90000             | 0.1 | sell | MID              | 50         |  8     | submission |
+      | lp1 | lp    | ETH/DEC19 | 90000             | 0.1 | sell | MID              | 50         |  7     | submission |
+      | lp1 | lp    | ETH/DEC19 | 90000             | 0.1 | sell | MID              | 50         |  2     | submission |
+
+    Then the parties place the following orders:
+      | party | market id | side | volume | price | resulting trades | type       | tif     | reference |
+      | aux   | ETH/DEC19 | buy  | 10     | 129   | 0                | TYPE_LIMIT | TIF_GTC | bestBid   |
+      | aux2  | ETH/DEC19 | sell | 10     | 155   | 0                | TYPE_LIMIT | TIF_GTC | bestOffer |
+      | aux   | ETH/DEC19 | buy  | 1      | 150   | 0                | TYPE_LIMIT | TIF_GTC |           |
+      | aux2  | ETH/DEC19 | sell | 1      | 150   | 0                | TYPE_LIMIT | TIF_GTC |           |
+    Then the opening auction period ends for market "ETH/DEC19"
+    And the market data for the market "ETH/DEC19" should be:
+      | mark price | trading mode            | auction trigger             | horizon | min bound | max bound |
+      | 150        | TRADING_MODE_CONTINUOUS | AUCTION_TRIGGER_UNSPECIFIED | 100     | 149       | 151       |
+
+    And the liquidity provisions should have the following states:
+      | id  | party | market    | commitment amount | status        |
+      | lp1 | lp    | ETH/DEC19 | 90000             | STATUS_ACTIVE |
+
+    # MID price is (129+155)/2=142, 
+    # 1st LP sell order should be at 142+16=158, but since that adjusted price would fall above the maximum bound of 151, it gets placed at that bound
+    # 2nd LP sell order should be at 142+8=150, since that's not above the max bound it should be unaffected
+    # 3rd LP sell order should be at 142+7=149, since that's not above the max bound it should be unaffected
+    # 4th LP sell order should be at 142+2=144, since that's not above the max bound it should be unaffected
+    # LP buy order should be at 140-20=120, but since reference price itself is below the min bound it should be placed at 139 (reference - minimum valid offset)
```

Comment:
```suggestion
    # LP buy order should be at 142-20=122, but since reference price itself is below the min bound it should be placed at 141 (reference - minimum valid offset)
```